Im facing this error on Kubuntu 20.04

federico@Kubuntu:~/plasma5-wallpapers-dynamic/build$ make
[  7%] Built target com.github.zzag.dynamic-wallpapers-metadata-json
[  7%] Automatic MOC for target kdynamicwallpaper
[  7%] Built target kdynamicwallpaper_autogen
[  7%] Building CXX object src/lib/CMakeFiles/kdynamicwallpaper.dir/kdynamicwallpaperreader.cpp.o
/home/federico/plasma5-wallpapers-dynamic/src/lib/kdynamicwallpaperreader.cpp:12:10: fatal error: QColorSpace: No existe el archivo o el directorio
   12 | #include <QColorSpace>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [src/lib/CMakeFiles/kdynamicwallpaper.dir/build.make:108: src/lib/CMakeFiles/kdynamicwallpaper.dir/kdynamicwallpaperreader.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:673: src/lib/CMakeFiles/kdynamicwallpaper.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
federico@Kubuntu:~/plasma5-wallpapers-dynamic/build$

--- a/src/lib/kdynamicwallpaperreader.cpp
+++ b/src/lib/kdynamicwallpaperreader.cpp
@@ -9,7 +9,9 @@
 
 #include <KLocalizedString>
 
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
 #include <QColorSpace>
+#endif
 #include <QFile>
 #include <QImage>
 
@@ -214,6 +216,7 @@ KDynamicWallpaperMetaData KDynamicWallpaperReaderPrivate::metaDataAt(int imageIn
     return metaData;
 }
 
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
 static QColorSpace colorProfileForImage(heif_image *image)
 {
     QColorSpace colorSpace;
@@ -234,6 +237,7 @@ static QColorSpace colorProfileForImage(heif_image *image)
 
     return colorSpace;
 }
+#endif
 
 QImage KDynamicWallpaperReaderPrivate::imageAt(int imageIndex)
 {
@@ -274,9 +278,11 @@ QImage KDynamicWallpaperReaderPrivate::imageAt(int imageIndex)
     auto cleanupFunc = [](void *data) { heif_image_release(static_cast<heif_image *>(data)); };
     QImage decodedImage(data, width, height, stride, format, cleanupFunc, image);
 
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
     const QColorSpace colorSpace = colorProfileForImage(image);
     if (colorSpace.isValid())
         decodedImage.setColorSpace(colorSpace);
+#endif
 
     heif_image_handle_release(handle); // heif_image will be destroyed by QImage
 

--- a/src/lib/kdynamicwallpaperwriter.cpp
+++ b/src/lib/kdynamicwallpaperwriter.cpp
@@ -9,7 +9,9 @@
 
 #include <KLocalizedString>
 
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
 #include <QColorSpace>
+#endif
 #include <QFile>
 #include <QImage>
 
@@ -226,6 +228,7 @@ bool KDynamicWallpaperWriterPrivate::write(const QImage &image, const KDynamicWa
     }
 
     if (options & KDynamicWallpaperWriter::PreserveColorProfile) {
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
         const QByteArray iccProfile = image.colorSpace().iccProfile();
         error = heif_image_set_raw_color_profile(heifImage, "prof", iccProfile.data(), iccProfile.size());
         if (error.code != heif_error_Ok) {
@@ -233,6 +236,9 @@ bool KDynamicWallpaperWriterPrivate::write(const QImage &image, const KDynamicWa
             errorString = i18n("Failed to set ICC profile: %1", error.message);
             goto error_image;
         }
+#else
+        qWarning() << "You need at least Qt 5.14 to preserve color profiles";
+#endif
     }
 
     data = heif_image_get_plane(heifImage, heif_channel_interleaved, &stride);

